test/system/system_test.go

```diff
@@ -580,8 +552,113 @@ var _ = Describe("Kratix", func() {
+
+				platform.eventuallyKubectl("get", "crd", "bash.test.kratix.io")
+				platform.eventuallyKubectl("get", "namespace", "promise-workflow-namespace")
+				worker.eventuallyKubectl("get", "namespace", "bash-dep-namespace-v1alpha1")
+				worker.eventuallyKubectl("get", "namespace", "bash-workflow-namespace-v1alpha1")
+			})
+
+			By("fulfiling the resource request", func() {
+				platform.kubectl("apply", "-f", exampleBashRequest(rrName))
+
+				By("executing the pipeline pod", func() {
+					platform.kubectl("wait", "--for=condition=PipelineCompleted", "bash", rrName, pipelineTimeout)
+				})
+
+				By("deploying the contents of /kratix/output to the appropriate destinations", func() {
+					platform.eventuallyKubectl("get", "namespace", "declarative-platform-only-rr-test")
+					worker.eventuallyKubectl("get", "namespace", "declarative-rr-test")
+				})
+			})
+
+			By("removing from the repo on resource delete", func() {
+				platform.kubectl("delete", "bash", rrName)
+
+				Eventually(func(g Gomega) {
+					g.Expect(platform.kubectl("get", "bash")).NotTo(ContainSubstring(rrName))
+					g.Expect(platform.kubectl("get", "namespace")).NotTo(ContainSubstring("imperative-rr-test"))
+					g.Expect(platform.kubectl("get", "namespace")).NotTo(ContainSubstring("declarative-platform-only-rr-test"))
+					g.Expect(worker.kubectl("get", "namespace")).NotTo(ContainSubstring("declarative-rr-test"))
+				}, timeout, interval).Should(Succeed())
+			})
+
+			By("removing from the repo on promise delete", func() {
+				platform.kubectl("delete", "promise", "bash")
+
+				Eventually(func(g Gomega) {
+					g.Expect(worker.kubectl("get", "namespace")).NotTo(ContainSubstring("bash-dep-namespace-v1alpha1"))
+					g.Expect(platform.kubectl("get", "namespace")).NotTo(ContainSubstring("promise-workflow-namespace"))
+					g.Expect(platform.kubectl("get", "promise")).ShouldNot(ContainSubstring("bash"))
+					g.Expect(platform.kubectl("get", "crd")).ShouldNot(ContainSubstring("bash"))
+				}, timeout, interval).Should(Succeed())
+			})
+		})
+	})
+
 })
 
+func exampleBashRequest(name string) string {
+	request := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "test.kratix.io/v1alpha1",
+			"kind":       "bash",
+			"metadata": map[string]interface{}{
+				"name": name,
+			},
+			"spec": map[string]interface{}{
+				"container0Cmd": `
+					set -x
+					if [ "${KRATIX_WORKFLOW_ACTION}" = "configure" ]; then :
+						echo "message: My awesome status message" > /kratix/metadata/status.yaml
+						echo "key: value" >> /kratix/metadata/status.yaml
+						mkdir -p /kratix/output/foo/
+						echo "{}" > /kratix/output/foo/example.json
+						kubectl get namespace imperative-$(yq '.metadata.name' /kratix/input/object.yaml) || kubectl create namespace imperative-$(yq '.metadata.name' /kratix/input/object.yaml)
+						exit 0
+					fi
+					kubectl delete namespace imperative-$(yq '.metadata.name' /kratix/input/object.yaml)
```

Comment:
better as an asset file?

Reply:
its one of those... i think this was an asset file before, then it became in-line... 🤔
(i just copied over from other test into a function so i could re-use)

Reply:
fair